### PR TITLE
Refactor: convert articleview into plain code

### PR DESCRIPTION
**Pull request by shenlebantongying**
The `articleview.ui` file is removed :)

It is converted to about 90 lines at the beginning of `ArticleWidget::ArticleWidget(`

Only those 90 lines matters. Other changes are just renaming. The clang-format kicks in if a line has name changes, thus this PR also includes some format changes.

You may compare this with the autogenerated `ui_articleview.h`. This new code is shorter and has no dark magic.

There is no behaviour changes in this PR.

### CODE REVIEW

File: articlewidget.hh [hunk 1]
```diff
@@ -429,36 +446,35 @@ class ResourceToSaveHandler: public QObject
...
   void done();
   void statusBarMessage( QString const & message, int timeout = 0, QPixmap const & pixmap = QPixmap() );
 
-public slots:
+ public slots:
   void downloadFinished();
 
-private:
+ private:
   std::list< sptr< Dictionary::DataRequest > > downloadRequests;
```
Comment (xiaoyifang) [inline articlewidget.hh:462]: seems the .clang-format file has configured wrong .

do not indent seems better.

Comment (shenlebantongying) [reply articlewidget.hh:461]: I fixed clang format, the problem is gone

**Comment by xiaoyifang** [comment]
a bold change 👍 

Since you  start the refactor, we can take a step more closer,  refactor fulltext search panel as a seperate widget ,so also the search panel.
(put in another PR is also welcome)

**Comment by shenlebantongying** [comment]
We cannot rename the articleView to articleWidget:

There are multiple places assuming the name is "view".

image

**Comment by xiaoyifang** [comment]
> We cannot rename the articleView to articleWidget:
> 
> There are multiple places assuming the name is "view". Rename will cause more confusion.
> 
> image

the method can also be renamed .

**Comment by shenlebantongying** [comment]
I reverted the articleview → articlewidget name change.

There are subtle places to assume this name, and may need another PR.

image

**Comment by shenlebantongying** [comment]
> refactor fulltext search panel as a seperate widget ,so also the search panel.

We should reorganize the codebase before doing that.

A structure like this is needed so that related widgets are grouped together.
```
src/
  articleView/
     webview.cc
     searchpanel.cc
     others.cc
```

See falkon's webtab folder which is a similar widget plus its components like articleview.cc.
https://github.com/KDE/falkon/tree/master/src/lib/webtab

**Comment by xiaoyifang** [comment]
one disadvantage of this PR is that the articleview is becoming even larger
ui auto generated techonolgy is a solution to group together the code logic and seperate the logic.

**Comment by xiaoyifang** [comment]
> QFrame, and in articleview.ui, the type of AriticalView is QWidget. What is this mismatch...

QFrame is subclass of  QWidget , this is ok.

**Comment by shenlebantongying** [comment]
> one disadvantage of this PR is that the articleview is becoming even larger

The problem is addressed. I pushed a new commit that puts sub-widgets into other files. 

(Refactoring plain code is extremely easy.)

The articleview.cpp only contains a few essential lines.

And it is pretty clean. A VBoxLayout that contains 3 widgets.

**Comment by xiaoyifang** [comment]
We may not be able to draw a conclusion between **gui** and **pure code**. They both have their area of expertise.

I can give this PR a try. Let me know when you finished.

**Comment by shenlebantongying** [comment]
This PR is finished. Other changes are all 1 to 1 name changes. I tested the interactions. There should be no regression.

File: articleview.ui [hunk 2]
```diff
@@ -1,258 +0,0 @@
...
-         <widget class="QToolButton" name="searchNext">
-          <property name="text">
-           <string>&amp;Next</string>
-          </property>
-          <property name="icon">
-           <iconset resource="resources.qrc">
-            <normaloff>:/icons/next.svg</normaloff>:/icons/next.svg</iconset>
-          </property>
-          <property name="shortcut">
-           <string>Ctrl+G</string>
```
Comment (shenlebantongying) [inline articleview.ui:187]: Yes, firefox & chrome defined Ctrl+G for the next.

In the meantime, the old code seems to have forgotten to add Ctrl+Shift+G.

**Pull request merged**